UncomplicatedCustomItems/API/Features/CustomItemHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Exiled.API.Features;
+using Exiled.API.Features.Items;
+using System;
+using System.Collections.Generic;
+using UncomplicatedCustomItems.Interfaces;
+
+namespace UncomplicatedCustomItems.API.Features
+{
+    internal class CustomItemHandler
+    {
+        public ICustomItem CustomItem { get; }
+
+        public Player Player { get; }
+
+        public Item Item { get; }
+
+        public ushort Serial { get; }
+
+        public Dictionary<ItemEvents, Func<CustomItemHandler, bool>> InternalEvents { get; }
+
+        public CustomItemHandler(ICustomItem customItem, Player player)
+        {
+            CustomItem = customItem;
+            Player = player;
+            Item = Item.Create(customItem.Item, player);
+            Serial = Item.Serial;
+            InternalEvents = new();
+        }
+
+        public void AddEventListener(ItemEvents Event, Func<CustomItemHandler, bool> Action)
+        {
+            InternalEvents[Event] = Action;
+        }
+
+        public void RemoveEventListener(ItemEvents Event)
+        {
+            InternalEvents.Remove(Event);
+        }
+
+        internal bool TriggerEvent(ItemEvents Event)
+        {
+            bool Allowed = true;
```

Comment:
var

Reply:
Does not make sense, is better to use bool